plugin-infra/go-plugin-access/src/com/thoughtworks/go/plugin/access/scm/SCMExtension.java

```diff
@@ -57,7 +56,7 @@ public SCMExtension(PluginManager pluginManager) {
         super(pluginManager, new PluginRequestHelper(pluginManager, goSupportedVersions, SCM_EXTENSION), SCM_EXTENSION);
         registerHandler("1.0", new PluginSettingsJsonMessageHandler1_0());
         messageHandlerMap.put("1.0", new JsonMessageHandler1_0());
-        registerJsonMessageHandlerForRequestProcessor("1.0", new JsonMessageHandlerForRequestProcessor1_0());
+        registerMessageHandlerForPluginSettingsRequestProcessor("1.0", new MessageHandlerForPluginSettingsRequestProcessor1_0());
```

Comment:
No need for this? SCMs do not have plugin settings, right?

Reply:
We have documented for SCMs and package repositories that plugin settings can be supported. Should any scm or package repo plugin make this call, we need to handle it appropriately.